src/api/routes/search.ts

```diff
@@ -0,0 +1,45 @@
+import * as express from 'express';
+import { addAsync, RouterWithAsync } from '@awaitjs/express';
+import { DataStore } from '../../datastore/common';
+import { isValidPrincipal } from '../../helpers';
+
+export function createSearchRouter(db: DataStore): RouterWithAsync {
+  const router = addAsync(express.Router());
+  router.getAsync('/:term', async (req, res) => {
+    const { term: rawTerm } = req.params;
+    const term = rawTerm.trim();
+
+    // Check if term is a 32-byte hash, e.g.:
+    //   `0x4ac9b89ec7f2a0ca3b4399888904f171d7bdf3460b1c63ea86c28a83c2feaad8`
+    //   `4ac9b89ec7f2a0ca3b4399888904f171d7bdf3460b1c63ea86c28a83c2feaad8`
+    let hashBuffer: Buffer | undefined;
+    if (term.length === 66 && term.toLowerCase().startsWith('0x')) {
+      hashBuffer = Buffer.from(term.slice(2), 'hex');
+    } else if (term.length === 64) {
+      hashBuffer = Buffer.from(term, 'hex');
+    }
+    if (hashBuffer !== undefined && hashBuffer.length === 32) {
```

Comment:
Is it possible that a valid contract principle could trigger this, when it shouldn't?

Reply:
I don't think so. At the very minimum, a contract principal has a `.` character (not valid hexadecimal).